HOMEWORK_30/FinalProject/Game.cpp

```diff
@@ -0,0 +1,435 @@
+#include "Game.h"
+#include <fstream>
+#include <cstdlib>
+#include <iostream>
+
+int readHighScore() 
+{
+    int highScore = 0;
+    std::ifstream file(RECORD_FILE);
+    if (file.is_open()) {
+        file >> highScore;
+        file.close();
+    }
+    return highScore;
+}
+
+void writeHighScore(int highScore) 
+{
+    std::ofstream file(RECORD_FILE);
+    if (file.is_open()) {
+        file << highScore;
+        file.close();
+    }
+}
+
+Game::Game()
+    : window(sf::VideoMode(720, 920), "Doodle Jump", sf::Style::Default, sf::ContextSettings{ 8 }),
+    state(GameState::Menu),
+    menuState(window),
+    pauseState(window),
+    gameOverState(window),
+    minHeightGap(window.getSize().y * 0.08f),
+    maxHeightGap(window.getSize().y * 0.1f),
+    dy(0),
+    gravity(0.25f),
+    jumpVelocity(-12),
+    isJumping(false),
+    maxHeight(window.getSize().y * 0.4f),
+    hasBounced(false),
+    score(0),
+    movingPlatformSpawnProbability(0.1f),
```

Comment:
Move these constants into separate `class Config` so they can be easily tweaked from one place. At some point you could move them out to file to make these changes without recompile step